### PR TITLE
Fix compile time warnings

### PR DESCRIPTION
**Pull request by sknat**
Small fixes to silence the compile time warnings

### CODE REVIEW

**Comment by MathiasRaoul** [comment]
+1  Do you plan to merge this PR?

File: src/cli.c [hunk 1]
```diff
@@ -176,7 +176,7 @@ static void send_header(quicly_stream_t *stream, int is_http1, int status, const
 
 static int flatten_file_vec(quicly_sendbuf_vec_t *vec, void *dst, size_t off, size_t len)
 {
-    int fd = (int)vec->cbdata;
+    int fd = (uint64_t)vec->cbdata;
```
Comment (kazuho) [inline src/cli.c]: Would you mind clarifying the warning that you are seeing here?

I do not see why casting into `uint64_t` then implicitly casting to `int` is better than directly casting to `int`.

File: t/test.c [hunk 2]
```diff
@@ -148,7 +148,7 @@ static int on_stream_open(quicly_stream_open_t *self, quicly_stream_t *stream)
 
 quicly_stream_open_t stream_open = {on_stream_open};
 
-static void test_vector(void)
+static __attribute__ ((unused)) void test_vector(void)
```
Comment (kazuho) [inline t/test.c]: I think we might prefer retaining this, because we should reenable this test at some time.

File: src/cli.c (hunk 1 above)
Comment (sknat) [reply src/cli.c]: I'm getting the following warning : 
`src/cli.c:179:14: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]` 
as vec->cbdata is a `void*` it's complaining to have to cast & change from 64bits to 32bits at the same time (on a 64bits system). But you're right, I might have gone a bit quick on that one, we would be better off with a `intptr_t`

btw I'm using `cmake version 3.10.2` and `Make 4.1` on a 64bit ubuntu and running `cmake . && make`

**Comment by kazuho** [comment]
Re `intptr_t`, thank you for the clarification. Makes sense. I have gone with using `uintptr_t` for casting, at the same time retaining the type of `fd` to be `int`, as it is my understanding that we do not have issues with casting between integer types of different size, and the type of a file descriptor is expected to be `int.`.

**Pull request closed without merging**